### PR TITLE
SONARIAC-1339 Decouple `HelmFileSystem` from `KubernetesParser`

### PR DESCRIPTION
**Pull request by navendu-barua-sonarsource**

### CODE REVIEW

**Review by nils-werner-sonarsource — CHANGES_REQUESTED** [review]
I acknowledge, that what was exactly written in the ticket was not doable, as since the ticket was created the code changed. However, the goal of the ticket was to decouple the HelmFileSystem from the Kubernetes Parser, to make a Proof-of-Concept I prepared this [PR](https://github.com/SonarSource/sonar-iac/pull/1193) so that you can have a quick look.

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/HelmProcessor.java [hunk 1]
```diff
@@ -85,6 +88,7 @@ String processHelmTemplate(String path, String source, HelmInputFileContext inpu
     inputFileContext.setSourceWithComments(sourceWithComments);
     inputFileContext.setAdditionalFiles(helmFilesystem.getRelatedHelmFiles(inputFileContext.inputFile));
     var fileContents = validateAndReadFiles(inputFileContext);
+    String path = helmFilesystem.getFileRelativePath(inputFileContext.inputFile);
```
[the remark this answers is not in the document]
Comment (navendu-barua-sonarsource) [reply iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/HelmProcessor.java:91]: which static method are you referring to here to replace `getFileRelativePath`? is it the static method `retrieveHelmProjectFolder (Path, FileSystem)`?

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/HelmProcessor.java [hunk 2]
```diff
@@ -85,7 +85,8 @@ String processHelmTemplate(String path, String source, HelmInputFileContext inpu
     inputFileContext.setSourceWithComments(sourceWithComments);
     inputFileContext.setAdditionalFiles(helmFilesystem.getRelatedHelmFiles(inputFileContext.inputFile));
     var fileContents = validateAndReadFiles(inputFileContext);
-    return evaluateHelmTemplate(path, inputFileContext, sourceWithComments, fileContents);
+    var path = HelmFileSystem.retrieveHelmProjectFolder(Path.of(inputFileContext.inputFile.uri()), inputFileContext.sensorContext.fileSystem());
+    return evaluateHelmTemplate(String.valueOf(path), inputFileContext, sourceWithComments, fileContents);
```
Comment (nils-werner-sonarsource) [inline iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/HelmProcessor.java:89]: Why do you need to apply `String.valueOf(path)` here?

Comment (navendu-barua-sonarsource) [reply iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/HelmProcessor.java:89]: evaluateHelmTemplate expects a String and not a Path as the first argument

File: iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/plugin/KubernetesParserTest.java [hunk 3]
```diff
@@ -519,17 +520,19 @@ void shouldNotIgnoreValuesYamlInTemplatesDirectory(String filename) throws URISy
 
   @Test
   void shouldIgnoreChartYaml() throws URISyntaxException {
-    when(helmFileSystem.retrieveHelmProjectFolder(any())).thenReturn(Path.of("/chart"));
-    when(inputFile.toString()).thenReturn("chart/Chart.yaml");
-    when(inputFile.filename()).thenReturn("Chart.yaml");
-    when(inputFile.uri()).thenReturn(new URI("file:///chart/Chart.yaml"));
-    when(inputFile.path()).thenReturn(Path.of("/chart/Chart.yaml"));
-    when(fileSystem.baseDir()).thenReturn(new File("/chart"));
+    try (var ignored = Mockito.mockStatic(HelmFileSystem.class)) {
```
Comment (nils-werner-sonarsource) [inline iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/plugin/KubernetesParserTest.java]: Please don't introduce static mocks. Rather you should challenge, why the method is static.

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/HelmProcessor.java (hunk 2 above)
Comment (nils-werner-sonarsource) [reply iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/HelmProcessor.java:88]: Please remove this as HelmFilesystem.getFileRelativePath returns a String, not a Path.

**Review by nils-werner-sonarsource — APPROVED** [review]

**Pull request merged**